### PR TITLE
Jackack/nand sim

### PR DESCRIPTION
**Pull request by Jackack**
Added ability to simulate multiple buggies. Due to topic name collision during sim, prepended buggy name to topics. Changed main and sim launch files to pass buggy name and is_sim into auto nodes. 

Tested simulation with two buggies inside wsl+docker. Todo: test updated sim code on short circuit.

### CODE REVIEW

**Comment by christianvluu** [comment]
Can you update documentation on how to use this sim too? (e.g. how to launch, interact, interface, etc.)

File: rb_ws/src/buggy/scripts/2d_sim/engine.py [hunk 1]
```diff
@@ -21,37 +22,39 @@ class Simulator:
     START_LONG = -79.9409643423245
     NOISE = True  # Noisy outputs for nav/odom?
 
-    def __init__(self, heading: float):
+    def __init__(self, starting_pose, velocity, buggy_name):
```
Comment (christianvluu) [inline rb_ws/src/buggy/scripts/2d_sim/engine.py:25]: typing

File: rb_ws/src/buggy/scripts/2d_sim/engine.py [hunk 2]
```diff
@@ -21,37 +22,39 @@ class Simulator:
...
         """
         Args:
             heading (float): degrees start heading of buggy
```
Comment (christianvluu) [inline rb_ws/src/buggy/scripts/2d_sim/engine.py:28]: functions headers

**Review by christianvluu — CHANGES_REQUESTED** [review]
And can you add screenshots? And also make a note in the documentation or in the git commit message when you merge that this does not contain the X11 visualization edits.

**Comment by Jackack** [comment]
Added all requested changes

**Review by christianvluu — APPROVED** [review]

**Pull request merged**